core/tests/webdriverio_utils/ContributorDashboardAdminPage.js

```diff
@@ -232,7 +236,147 @@ var ContributorDashboardAdminPage = function () {
+    because the mat-datepicker that we use here do not contain any
+    such classes for e2e test purposes. Also we can not add any e2e
+    test class to any ansector element since, the ancestor to this
+    is the body tag, in the DOM.
+    **/
+    var nextMonthButton = $('.mat-calendar-next-button');
+    var prevMonthButton = $('.mat-calendar-previous-button');
+    var calenderBody = $('.mat-calendar-content')
+      .$('.mat-calendar-table')
+      .$('.mat-calendar-body')
+      .$('tr')
+      .$('td');
+
+    var months = [
+      'JAN',
+      'FEB',
+      'MAR',
+      'APR',
+      'MAY',
+      'JUN',
+      'JUL',
+      'AUG',
+      'SEP',
+      'OCT',
+      'NOV',
+      'DEC',
+    ];
+
+    var monthText = await calenderBody.getText();
+    var monthIndex;
+    for (let i = 0; i < 12; i++) {
+      if (months[i] === monthText) {
+        monthIndex = i;
+      }
+    }
+
+    if (numberOfMonths > 0) {
+      for (let i = 0; i < numberOfMonths; i++) {
+        await action.click('Next Month Button', nextMonthButton);
+        if (monthIndex === 11) {
+          monthIndex = 0;
+        } else {
+          monthIndex++;
+        }
+        monthText = await calenderBody.getText();
+        expect(months[monthIndex]).toBe(monthText);
+      }
+    } else if (numberOfMonths < 0) {
+      for (let i = 0; i < Math.abs(numberOfMonths); i++) {
+        await action.click('Previous Month Button', prevMonthButton);
+        if (monthIndex === 0) {
+          monthIndex = 11;
+        } else {
+          monthIndex--;
+        }
+        monthText = await calenderBody.getText();
+        expect(months[monthIndex]).toBe(monthText);
+      }
+    }
+  };
+
+  this.selectDate = async function (
+    datePickerToggle,
+    datePickerInput,
+    selectedDate
+  ) {
+    var initialValueOfDatePicker = await action.getValue(
+      'Date Picker Input',
+      datePickerInput
+    );
+
+    await action.click('Date Picker Toggle', datePickerToggle);
+
+    // Here selectedDate is a Date object that is to be selected by date picker.
+    var day = selectedDate.getDate();
+    var month = selectedDate.getMonth() + 1;
+    var year = selectedDate.getFullYear();
+
+    if (year) {
+      var yearsToNavigate =
+        year - new Date(initialValueOfDatePicker).getFullYear();
+
+      await this.navigateThroughMonths(yearsToNavigate * 12);
+    }
+
+    if (month) {
+      var currentMonth = new Date(initialValueOfDatePicker).getMonth() + 1;
+      var monthsToNavigate = month - currentMonth;
+
+      await this.navigateThroughMonths(monthsToNavigate);
+    }
+
+    if (day) {
+      /**
+    We require to use element selector classes without "e2e-*" fix
+    because the mat-datepicker that we use here do not contain any
+    such classes for e2e test purposes. Also we can not add any e2e
+    test class to any ansector element since, the ancestor to this
+    is the body tag, in the DOM.
+    **/
```

Comment:
Please fix the indentation. Also, there are some grammar issues. Here is the comment with those fixed:

> We need to use element selector classes without the "e2e-" prefix because the mat-datepicker that we use here does not contain any such classes. Since the datepicker is a third-party dependency, we cannot edit it to add "e2e-" classes. We cannot use an ancestor element with the "e2e-" class because the datepicker's ancestor is the body tag.

ditto above